### PR TITLE
chore: update readme

### PR DESCRIPTION
**Pull request by wreality**
This PR:

* Updates documentation site to vitepress.
* Expands documentation to include installation
* Updates documentation content
* Makes team list all-contributors compatible
* Updates README with installation and contributing references.

### CODE REVIEW

**Comment by wreality** [comment]
@all-contributors please add @wreality for code, mentoring, infrastructure and review

**Comment by wreality** [comment]
Ready for review!  For rendered previews see the netlify deploy linked above.

**Review by gmeben — CHANGES_REQUESTED** [review]
This is a very beautiful-looking update to the documentation. I have some suggestions. 

I was not able to get the `docs` site working locally following the existing instructions: https://deploy-preview-1988--eloquent-galileo-3be94a.netlify.app/developers/documentation.html#serving-test-docs-using-lando

File: docs/install/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+APP_PORT=8888  # <-- Set to port the application should be exposed on.
+
+DB_HOST=127.0.0.1 # <-- Enter your database credentials in these fields
+DB_PORT=3306
+DB_DATABASE=laravel
+DB_USERNAME=root
+DB_PASSWORD=
+
+MAIL_FROM_ADDRESS=<valid email address> # <-- This MUST be a valid email address
+MAIL_FROM_NAME="Pilcrow Mailer"   # <-- Supply a name to be used as the sender for mail
```
[the remark this answers is not in the document]
Comment (wreality) [reply docs/install/index.md]: Resolved in commit

**Comment by wreality** [comment]
> I was not able to get the `docs` site working locally following the existing instructions: https://deploy-preview-1988--eloquent-galileo-3be94a.netlify.app/developers/documentation.html#serving-test-docs-using-lando

This is fixed now.  I neglected to move the lando config over from the old docs directory.  Everything should be address now.

**Review by gmeben — APPROVED** [review]
I encounter a 404 when I attempt to visit the Team page, but all the other pages load and look fine. I'll attribute that to something missing locally because it loads in the deployment preview. That can be troubleshooted later. 

Screenshot of the Team page with Chrome's developer tools open. An error message for contributors.data.ts reads, 'The 'path' argument must be of type string or an instance of Buffer or URL. Received undefined'

**Pull request merged**